test/libsolidity/semanticTests/interfaceID/interfaces.sol

```diff
@@ -0,0 +1,67 @@
+interface HelloWorld {
+    function hello() external pure;
+    function world(int) external pure;
+}
+
+interface HelloWorldDerived is HelloWorld {
+    function other() external pure;
+}
+
+interface ERC165 {
+    /// @notice Query if a contract implements an interface
+    /// @param interfaceID The interface identifier, as specified in ERC-165
+    /// @dev Interface identification is specified in ERC-165. This function
+    ///  uses less than 30,000 gas.
+    /// @return `true` if the contract implements `interfaceID` and
+    ///  `interfaceID` is not 0xffffffff, `false` otherwise
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+
+contract Test {
+    bytes4 public ghello_world_interfaceId = type(HelloWorld).interfaceId;
+    bytes4 public ERC165_interfaceId = type(ERC165).interfaceId;
+
+    function hello() public pure returns (bytes4 data){
+        HelloWorld i;
+        return i.hello.selector;
```

Comment:
`HelloWorld.hello.selector` does not work?

Reply:
Ah yes, it seem to work. At least if Yul is not used. If Yul is used, we got the following error:
```
Exception during extracted test: /Users/alex/git/solidity/libsolidity/codegen/ir/IRGeneratorForStatements.cpp(988): Throw in function virtual void solidity::frontend::IRGeneratorForStatements::endVisit(const solidity::frontend::MemberAccess &)
Dynamic exception type: boost::wrapexcept<solidity::langutil::InternalCompilerError>
std::exception::what: Member access to unknown type.
[solidity::util::tag_comment*] = Member access to unknown type.
```

Reply:
https://github.com/ethereum/solidity/pull/8684